qcodes/utils/update/qcodes_updater.py

```diff
@@ -0,0 +1,110 @@
+import os
+import subprocess
+import sys
+import shutil
+import datetime
+import warnings
+import git
+
+import qcodes
+
+def qcodes_backup(env_name: str = 'qcodes', env_backup_name: str = 'qcodes_backup'):
+    """
+    Back up QCoDeS Anaconda Python environment.
+
+    Args:
+        env_name: Name of the environment to be updated
+        env_backup_name: Name of the back up environment to be created
+    """
+    _source = os.sep.join(qcodes.__file__.split(os.sep)[:-2])
+
+    # Make sure that git working tree is clean
+    repo = git.Repo(_source)
+    if repo.is_dirty():
+        print("QCoDeS working tree is not clean. Please commit or stash your changes, and try again.")
+        sys.exit()
```

Comment:
Since this is a function and not a script I think it should just return

Reply:
Yes, in case the back up function called directly, a simple return would suffice. However, when this one called from one of the update functions (they are designed to do so), and If the tree is dirty, it will return None, and an unnecessary error will raise as update functions expects 4 strings as return values. I think code should terminate if there is no possibility for an update?

Reply:
I think that reasoning is a bit upside-down. It's the responsibility of whoever calls the function to correctly handle the different outcomes of that call, not the responsibility of the function itself to try to anticipate where and how it is going to get called.

In this case, I'd suggest that the update function raises if the backup fails. Or, alternatively, that the backup function raises if the tree is not clean. The update function can then catch that error and re-raise it with some context.